tests/specs/examples/erc721-spec.md

```diff
@@ -0,0 +1,147 @@
+ERC721-ish Verification
+=======================
+
+```k
+requires "edsl.md"
+requires "optimizations.md"
+requires "lemmas/lemmas.k"
+requires "lemmas/infinite-gas.k"
+```
+
+Solidity Code
+-------------
+
+File [`ERC721.sol`](ERC721.sol) contains some code snippets we want to verify the functional correctness of.
+Call `kevm solc-to-k ERC721.sol ERC721 > erc721-bin-runtime.k`, to generate the helper K files.
+
+Verification Module
+-------------------
+
+Helper module for verification tasks.
+
+-   Add any lemmas needed for our proofs.
+-   Import a large body of existing lemmas from KEVM.
+
+```k
+requires "erc721-bin-runtime.k"
+
+module VERIFICATION
+    imports EDSL
+    imports LEMMAS
+    imports INFINITE-GAS
+    imports EVM-OPTIMIZATIONS
+    imports ERC721-BIN-RUNTIME
+
+    syntax Step ::= ByteArray | Int | Bool
+    syntax KItem ::= runLemma ( Step ) | doneLemma ( Step )
+ // -------------------------------------------------------
+    rule <k> runLemma(S) => doneLemma(S) ... </k>
+
+ // Lemmas
+ // ---------------
+
+    rule         X &Int 127 <Int 128 => true requires 0 <=Int X                   [simplification, smt-lemma]
+    rule 0 <=Int X &Int 127          => true requires 0 <=Int X                   [simplification, smt-lemma]
+ // rule         X &Int 127 <=Int X  => true requires 0 <=Int X                   [simplification, smt-lemma]
+    rule         X &Int 127  <Int 32 => true requires 0 <=Int X andBool X <Int 32 [simplification, smt-lemma]
+
+endmodule
+```
+
+K Specifications
+----------------
+
+Formal specifications (using KEVM) of the correctness properties for our Solidity code.
+
+```k
+module ERC721-SPEC
+    imports VERIFICATION
+```
+
+```k
+claim <k> runLemma(chop(chop(chop(chop(chop(chop((#lookup(ACCT_STORAGE, 0) /Int 2 &Int 127) +Int 31) /Int 32 *Int 32) +Int 32) +Int 128) +Int 32) +Int 32))
+       => doneLemma((((((((#lookup(ACCT_STORAGE, 0) /Int 2 &Int 127) +Int 31) /Int 32 *Int 32) +Int 32) +Int 128) +Int 32) +Int 32)) </k>
+```
+
+```k
+claim <k> runLemma(#lookup(ACCT_STORAGE, 0) /Int 2 <Int 32)
+       => doneLemma(false) </k>
+ requires 32 <=Int #lookup(ACCT_STORAGE, 0) /Int 2 &Int 127
```

Comment:
`````suggestion
claim <k> runLemma(chop(chop(chop(chop(chop(chop((#lookup(ACCT_STORAGE, 0) /Int 2 &Int 127) +Int 31) /Int 32 *Int 32) +Int 32) +Int 128) +Int 32) +Int 32))
       => doneLemma((((((((#lookup(ACCT_STORAGE, 0) /Int 2 &Int 127) +Int 31) /Int 32 *Int 32) +Int 32) +Int 128) +Int 32) +Int 32)) ... </k>
```

```k
claim <k> runLemma(#lookup(ACCT_STORAGE, 0) /Int 2 <Int 32)
       => doneLemma(false) ... </k>
 requires 32 <=Int #lookup(ACCT_STORAGE, 0) /Int 2 &Int 127
`````

Reply:
Also, I'm surprised that `... +Int 32) +Int 128) +Int 32) ...` doesn't get simplified out. Maybe you need a lemma like `(X +Int Y) +Int Z) => X +Int (Y +Int Z) [simplification, concrete(Y, Z)]`? But I guess we should already have lemmas for dealing with this in `INT-SIMPLIFICATION` in `tests/specs/int-simplification.k`, which should be included via `INFINITE-GAS`.

Or maybe this simplification is actually happenning, but your proof claim RHS isn't as simplified as it can be?